### ISSUE TITLE
localhost:3000 no longer working

### ISSUE THREAD

**Issue by zhushuihua** [supabase/supabase#12003]
xxx.xxxx.xxx.xxx/3000 is messy with latest versions. 
All directs to https.

**Comment by ruggi99** [comment]
Can you be a little more specific?
It's not clear what you mean

**Comment by AVanVlack** [comment]
I believe I am seeing this same issue. Studio is prefixing the URLs for all resource with https, instead of http, resulting in  SSL errors for all resources after the initial page request.

This is happening for me on a new docker deployment of the latest build. The only change was the ```SUPABASE_PUBLIC_URL=http://192.168.1.11:3000``` in .env

![image](https://user-images.githubusercontent.com/3699695/215322305-7344910f-e82c-4644-ad8d-5957e115c697.png)

**Comment by ruggi99** [comment]
@AVanVlack Hi,
please post a screenshot of this path: `http://172.16.42.52:3000/api/props/project/default/api`.
Thanks

**Comment by ruggi99** [comment]
From what I see from the screenshot it seems an issue with SSL on the frontend, not backend.
It can't load fonts or needed javascript to work properly.
Seems like you tried to add SSL on frontend but it is misconfigured. Supabase self-hosted provides no SSL support AFAIK.
Can you please describe how you setupped Supabase self-hosted?

Thanks

**Comment by AVanVlack** [comment]
Here is props route with defaults:

![image](https://user-images.githubusercontent.com/3699695/215324202-79e622fd-d834-40b2-b675-c970813632b1.png)

The client is directly requesting from the server on the hosting port. No SSL proxy or anything in-between.

Instructions where followed from self host guide:
```
git clone --depth 1 https://github.com/supabase/supabase
cd supabase/docker
cp .env.example .env
docker compose up 
```

This setup was working with a build about 5 months old before removing all everything and reinstalling.

**Comment by AVanVlack** [comment]
Ahh I think I see the issue. The initial request is sending a ```upgrade-insecure-requests``` on the ```Content-Security-Policy``` header:

![image](https://user-images.githubusercontent.com/3699695/215325551-184911f6-226b-450f-b0f2-ae571e43b664.png)

It looks like it was added in #11932

**Comment by ruggi99** [comment]
Graat work!
I think you found it.
Will send PR disabling it for self-host

Thanks

**Comment by AVanVlack** [comment]
Thank you!

**Comment by zhushuihua** [comment]
The issue is still there. Why it is now closed?

**Comment by ruggi99** [comment]
The issue seems to be fixed but it is not deployed yet.
A maintainer should build the image, upload to Docker Hub and include in `docker-compose.yml`
By now you have two alternatives:
- Build locally yourself studio image
- Downgrade pre-built studio image to at least one week ago.

**Comment by sweatybridge** [comment]
The fix is released in latest studio build [20230127-6bfd87b](https://hub.docker.com/layers/supabase/studio/20230127-6bfd87b/images/sha256-17ca7066ce0eacf285e36e423660fc8ada8735cfe59f6bcc5046118f9668c153?context=explore)

You can update the image version used by docker-compose file here https://github.com/supabase/supabase/blob/master/docker/docker-compose.yml#L12. If you are happy with the fix, please consider sending a PR. I would appreciate your help as I cannot merge my own PR.

**Comment by ruggi99** [comment]
Interesting the wrong timestamp, it confused me. I didn't see Docker Hub telling me it was uploaded 4 hours ago.
Will send a PR.
